apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/EndpointJsonComparatorBase.java

```diff
@@ -0,0 +1,163 @@
+   */
+  public void assertJsonDiffIsEmpty(String endpointId, String endpointResponse) {
+    String expectedJson =
+        ServerTestUtils.readFile(
+            ServerTestUtils.generateEndpointJsonFileName(getExpectedJsonResponseDir(), endpointId));
+    AssertUtils.assertJsonEquals(
+        expectedJson, endpointResponse, ServerTestUtils.JSON_COMPARE_IGNORED_PATHS);
+  }
+
+  /**
+   * Gets the json response for the specified endpoint request string and default content type json
+   * header.
+   *
+   * @param requestString the request string
+   * @return the json response for
+   */
+  protected static String getJsonResponseFor(String requestString) {
+    return getJsonResponseFor(
+        requestString, new Headers(new Header("Content-Type", "application/json+fhir")));
+  }
+
+  /**
+   * Gets the json response for the specified endpoint request string and specified headers.
+   *
+   * @param requestString the request string
+   * @param headers the headers for the request
+   * @return the json response for
+   */
+  protected static String getJsonResponseFor(String requestString, Headers headers) {
+    return given()
+        .spec(requestAuth)
+        .headers(headers)
+        .expect()
+        .log()
+        .ifError()
+        .statusCode(200)
+        .when()
+        .get(requestString)
+        .then()
+        .extract()
+        .response()
+        .asString();
+  }
+
+  /**
+   * Gets the expected json response directory. These are generated as a point in time comparison
+   * when we know the responses are in a good state so that the json response tests can regression
+   * test against them.
+   *
+   * <p>Should be overridden in the version specific class.
+   *
+   * @return the expected json response dir
+   */
+  protected abstract Path getExpectedJsonResponseDir();
+
+  /**
+   * Replaces ignored fields with filler text.
+   *
+   * @param parent the {@link JsonNode} on which to perform the replacement
+   * @param fieldName the {@link String} name of the field that is being replaced
+   * @param pattern an optional {@link Pattern} pattern to correctly identify fields needing to be
+   *     replaced
+   */
+  protected static void replaceIgnoredFieldsWithFillerText(
+      JsonNode parent, String fieldName, Optional<Pattern> pattern) {
+    if (parent.has(fieldName)) {
+      if (pattern.isPresent()) {
+        Pattern p = pattern.get();
+        Matcher m = p.matcher(parent.get(fieldName).toString());
+        if (m.find())
+          if (fieldName.equals("url")) {
+            // Only replace the port numbers (m.group(2)) on urls
+            String replacementUrl = m.group(1) + IGNORED_FIELD_TEXT + m.group(3);
+            ((ObjectNode) parent)
+                .put(fieldName, replacementUrl.substring(0, replacementUrl.length() - 1));
+          } else ((ObjectNode) parent).put(fieldName, IGNORED_FIELD_TEXT);
+      } else ((ObjectNode) parent).put(fieldName, IGNORED_FIELD_TEXT);
```

Comment:
Please use braces here.

Reply:
Good catch, this is copied old code. Fixed it!